_posts/2019-02-28-into-the-darkness-of-viewmodel-lists-composition.md

```diff
@@ -66,18 +68,18 @@ The Marketing `RequestHandler` publishes an in-memory event to notify that some
 
 ![1551362519284](/img/posts/into-the-darkness-of-viewmodel-lists-composition/1551362519284.png){:class="img-responsive"}
 
-Once all components have received the event they go to their respective backends to retrieve all the information related to all the loaded products, in 1 single roundtrip. 
+Once all components have received the event they go to their respective backends to retrieve the information related to all the loaded products, in a single roundtrip. 
 
 ![1551362531125](/img/posts/into-the-darkness-of-viewmodel-lists-composition/1551362531125.png){:class="img-responsive"}
 
-Once all the information are retrieved the last step, the composition, is not that different from the Single Item Composition scenario:
+Once the information has been retrieved, the last step (i.e. the composition) is not that different from the Single Item Composition scenario:
 
 ![1551362461054](/img/posts/into-the-darkness-of-viewmodel-lists-composition/1551362461054.png){:class="img-responsive"}
 
 It's as simple as an in-memory `foreach` iterating over the loaded product, giving each interested party the opportunity to augment each product with their data.
 
 ## Conclusions
 
-Composing lists of items sounds complex and scary, we don't want to fall into the `SELECT N+1 trap over HTTP`. Looking deeply into the problem however, first we can identify who is the service responsible to get the list of items to display keys. Once we have all the keys we can apply an approach similar to the one used for the Single Item Composition and retrieve data in batches to optimize server-side communication.
+Composing lists of items sounds complex and scary; we don't want to fall into the `SELECT N+1 trap over HTTP`. Looking deeply into the problem however, first we can identify who is the service responsible to get the list of items to display keys. Once we have all the keys we can apply an approach similar to the one used for the Single Item Composition and retrieve data in batches to optimize server-side communication.
```

Comment:
This change is probably on the pedantic side and could be ignored. Semi-colons are weird things and I don't fully understand when and how they're used. I felt that these were two separate statements so a comma doesn't really fit. But they're related which is why I went with a semi-colon.

Reply:
😊